src/main/java/org/jabref/gui/autosaveandbackup/BackupManager.java

```diff
@@ -1,384 +1,705 @@
-                if (result) {
-                    LOGGER.info("Backup file {} differs from current file {}", latestBackupPath, originalPath);
-                }
-                return result;
-            } catch (IOException e) {
-                LOGGER.debug("Could not compare original file and backup file.", e);
-                // User has to investigate in this case
-                return true;
-            }
-        }).orElse(false);
+
+    void startBackupTask(Path backupDir, BibDatabaseContext bibDatabaseContext) {
+        LOGGER.info("Initializing backup task for directory: {} and file: {}", backupDir, bibDatabaseContext.getDatabasePath().orElseThrow());
+        executor.scheduleAtFixedRate(
+                () -> {
+                    try {
+                        Path dbFile = bibDatabaseContext.getDatabasePath().orElseThrow(() -> new IllegalArgumentException("Database path is not provided."));
+                        // copyDatabaseFileToBackupDir(dbFile, backupDir);
+                        performBackup(dbFile, backupDir);
+                    } catch (IOException | GitAPIException e) {
+                        LOGGER.error("Error during backup", e);
+                    }
+                },
+                DELAY_BETWEEN_BACKUP_ATTEMPTS_IN_SECONDS,
+                DELAY_BETWEEN_BACKUP_ATTEMPTS_IN_SECONDS,
+                TimeUnit.SECONDS);
+        LOGGER.info("Backup task scheduled with a delay of {} seconds", DELAY_BETWEEN_BACKUP_ATTEMPTS_IN_SECONDS);
     }
 
     /**
-     * Restores the backup file by copying and overwriting the original one.
+     * Performs the backup by checking for changes and committing them to the Git repository.
      *
-     * @param originalPath Path to the file which should be equalized to the backup file.
+     * @param backupDir the backup directory
+     * @param dbfile the database file
+     * @throws IOException     if an I/O error occurs
+     * @throws GitAPIException if a Git API error occurs
      */
-    public static void restoreBackup(Path originalPath, Path backupDir) {
-        Optional<Path> backupPath = getLatestBackupPath(originalPath, backupDir);
-        if (backupPath.isEmpty()) {
-            LOGGER.error("There is no backup file");
+
+    protected void performBackup(Path dbfile, Path backupDir) throws IOException, GitAPIException {
+
+        boolean needsCommit = backupGitDiffers(dbfile, backupDir);
+
+        if (!needsBackup && !needsCommit) {
+            LOGGER.info("No changes detected, beacuse needsBackup is :{} and needsCommit is :{}", needsBackup, needsCommit);
             return;
         }
-        try {
-            Files.copy(backupPath.get(), originalPath, StandardCopyOption.REPLACE_EXISTING);
-        } catch (IOException e) {
-            LOGGER.error("Error while restoring the backup file.", e);
-        }
+
+        if (needsBackup) {
+            LOGGER.info("Backup needed, because needsBackup is :{}", needsBackup);
```

Comment:
```suggestion
            LOGGER.info("Backup needed, because needsBackup is: {}", needsBackup);
```

Please use Amercian positioning of spaces and colon: First comes the colon, then the space.